enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1158 @@
+Tactically, `strategy: Disabled` provides a convenient syntax for bypassing any configured 
+change management policy without permanently deleting its configuration.
+
+For example, if SRE needs to apply emergency corrective action on a cluster with a `MaintenanceSchedule` change
+management strategy configured, they can simply set `strategy: Disabled` without having to delete the existing
+`maintenanceSchedule` stanza which configures the previous strategy. Once the correct action has been completed,
+SRE simply restores `strategy: MaintenanceSchedule` and the previous configuration begins to be enforced.
+
+Configurations for multiple management strategies can be recorded in the `config` stanza, but
+only one strategy can be active at a given time.
+
+Each strategy will support a policy for pausing or unpausing (permitting) material changes from being initiated. 
+This will be referred to as the strategy's enforcement state (or just "state"). The enforcement state for a
+strategy can be either "paused" or "unpaused" (a.k.a. "permissive"). The `Disabled` strategy enforcement state 
+is always permissive -- allowing material changes to be initiated (see [Change Management
+Hierarchy](#change-management-hierarchy) for caveats).
+
+All change management strategies, except `Disabled`, are subject to the following `changeManagement` fields:
+- `changeManagement.disabledUntil: <bool|date>`: When `disabledUntil: true` or `disabledUntil: <future-date>`, the interpreted strategy for 
+  change management in the resource is `Disabled`. Setting a future date in `disabledUntil` offers a less invasive (i.e. no important configuration needs to be changed) method to 
+  disable change management constraints (e.g. if it is critical to roll out a fix) and a method that
+  does not need to be reverted (i.e. it will naturally expire after the specified date and the configured
+  change management strategy will re-activate).
+- `changeManagement.pausedUntil: <bool|date>`: Unless the effective active strategy is Disabled, `pausedUntil: true` or `pausedUntil: <future-date>`, change management must 
+  pause material changes.
+
+### Change Management Status
+Change Management information will also be reflected in resource status. Each resource 
+which contains the stanza in its `spec` will expose its current impact in its `status`. 
+Common user interfaces for aggregating and displaying progress of these underlying resources
+should be updated to proxy that status information to the end users.
+
+### Change Management Metrics
+Cluster wide change management information will be made available through cluster metrics. Each resource
+containing the stanza should expose the following metrics:
+- The number of seconds until the next known permitted change window. 0 if changes can currently be initiated. -1 if changes are paused indefinitely. -2 if no permitted window can be computed.
```

Comment:
My motivation was to simplify the promql necessary for useful alerting by an ops team. A simple rule might be `warn "Invalid update schedule" when seconds_until_update < 0 or seconds_until_update_permitted > 60 days` vs enumerating booleans for why things might be busted. SD would certainly know best here, so I'm open to alternatives. I just want any rule to survive the introduction of new error conditions.

Reply:
From the observability perspective it does reduce cardinality to use the value to encode the state.  As noted it will work.  It gets a bit brittle when you now have a mapping through various layers and have to know that encoded meaning in the metric value.